darwin/cli_functions.py

```diff
@@ -883,6 +881,59 @@ def convert(format: str, files: List[PathLike], output_dir: Path) -> None:
     export_annotations(parser, files, output_dir)
 
 
+def post_comment(
+    dataset_slug: str, filename: str, text: str, x: float = 1, y: float = 1, w: float = 1, h: float = 1
+) -> None:
+    """
+    Creates a comment box with a comment for the given file in the given dataset.
+
+    Parameters
+    ----------
+    dataset_slug: str
+        The slug of the dataset the item belongs to.
+    filename: str
+        The filename to receive the commment.
+    text: str
+        The comment.
+    x: float, default: 1
+        X value of the top left coordinate for the comment box.
+    y: float, default: 1
+        Y value of the top left coordinate for the comment box.
+    w: float, default: 1
+        Width of the comment box.
+    h: float, default: 1
+        Height of the comment box.
+
+    Raises
+    ------
+    NotFound
+        If the Dataset was not found.
+    """
+    client: Client = _load_client(dataset_identifier=dataset_slug)
+    console = Console()
+
+    try:
+        dataset = client.get_remote_dataset(dataset_identifier=dataset_slug)
+    except NotFound:
+        _error(f"unable to find dataset: {dataset_slug}")
+
+    items: List[DatasetItem] = list(dataset.fetch_remote_files(filters={"filenames": [filename]}))
+
+    if len(items) == 0:
+        console.print(f"[bold yellow]No files matching '{filename}' found...")
+    else:
+        for item in items:
+            maybe_workflow_id: Optional[int] = item.current_workflow_id
+
+            if maybe_workflow_id is None:
+                workflow_id: int = client.instantitate_item(item.id)
+            else:
+                workflow_id = maybe_workflow_id
+
+            client.post_workflow_comment(workflow_id, text, x, y, w, h)
```

Comment:
Can this be unsuccessful? If so, we could either:
- collect successful and unsuccessful operations, then print a final message
- simply print an error message for unsuccessful posts

Reply:
Yes, the `post_workflow_comment` can be unsuccessful, the HTTP call can fail (in which case we will raise with the error code, or the result may not be what we expect (in which case I will raise with `ValueError`). But as I mentioned, we will ever only execute this once (even though it is a for loop).

Reply:
Fair enough about that `for` point, but if the request in unsuccessful, we could catch the exception here and print a nice error message

Reply:
fixed